### ISSUE TITLE
Interpolation maxes out one core & the other processes & cores are idle

### ISSUE THREAD

**Issue by dustyny** [pelias/dockerfiles#37]
In my pelias.json file I've configured it to import the state of Florida (USA). The virtual machine has 4 cores but only 1 seems to be utilized. 

I see multiple threads running but most of them are idle. So this seems like a bug but perhaps it's expected behavior that hasn't been documented (or I didn't see it in the docs). 

Is this correct or can I make a change that will distributed the load across all the threads & cores? I'd love to use 36 or 72 cores and get this processing window down as small time as possible. 

Thank you

<img width="1288" alt="by default 2017-11-07 at 6 15 52 am" src="https://user-images.githubusercontent.com/7416333/32493777-b5b0d132-c384-11e7-97d5-aa947e307e83.png">

**Comment by orangejulius** [comment]
Hey @dustyny,
Thanks for reporting this. This is expected, but not desired. Our interpolation import process runs only a single process currently. Node.js can't really run multiple CPU intensive tasks at once from within a single process, so this is the normal CPU profile of almost any computation heavy Node.js process. 

However, there are a lot of opportunities to parellelize the interpolation import by processing multiple files simultaneously in multiple processes. Most OpenAddresses files which are the bulk of what's imported by the interpolation engine contain data at the county level, which even for just Florida could get you to about 36 cores working at once (you might need a bit of RAM ;) ).

This is something we want to work on soon and if you'd like to help out we can definitely guide you in the right direction. It would involve some javascript, sqlite, and almost certainly some bash scripting :)

**Comment by dustyny** [comment]
@orangejulius thank you for the quick reply. Totally makes sense. 

I'm struggling trying to figure out how to configure Pelias, populate it with data & make it production ready. The documentation doesn't say much on the subject. Can you tell me what you think best practice would be for what I'm trying to accomplish. I'll do my best to document it in a doc/blog post.

I'll be using Pelias as a secondary source of truth in my address correction process. Our data pipeline processes data from "jurisdictions" in a state which can cross county lines, contain multiple cities, etc. This will live in our pipeline and I can run it as a service calling it via REST or I can wrap it in a python script and run it as a batch process.

So would I have to spin up an instance of pelias for each state or is there a way to build a single instance that holds the entire United States?

**Comment by missinglink** [comment]
re: the original issue https://github.com/pelias/interpolation/pull/94

**Comment by dianashk** [comment]
@dustyny, you can certainly populate a single instance of Pelias to cover the entire world or several regions. You would need to run the importers a few different times to import regions with different hierarchies, but if you wanted to get the entire US, that would be entirely possible. Check out our [pelias/dockerfiles](https://github.com/pelias/dockerfiles] repository for more up-to-date information about Pelias setup and deployment.

**Comment by dustyny** [comment]
@dianashk I followed the tutorial on how to get one city setup but that doesn't talk about running importers. 

The documentation is not clear on how this works. For instance I don't  know if I have to manually download the files and put the in the data directories or if the sh scripts handle this for me. Some parts of the documentation say to download others say it auto downloads (like when passing in an empty array to openaddresses). 

We really could use a list of steps that to take. Even if it's just 8 sentences that would really help unblock me. I'm 40+ hours on this and it's getting harder to carve out time.

**Comment by dianashk** [comment]
@dustyny, I'm glad you found the [tutorial](https://github.com/pelias/dockerfiles/blob/master/how_to_guide.pdf)! It was intended to explain exactly how to go through the build process followed by deploying the API and relevant microservices and test each step.

Any commands in the tutorial that use the `openstreetmap`, `openaddresses`, and `whosonfirst` docker images are performing steps in the build process. `placeholder` and `interpolation` also have build steps, but then are part of the deployed Pelias API system, as they are also microservices.

Since you've closed this issues, I'm hopeful that you got things working. But if not, we'd be happy to set up a call to help you make progress.

Cheers!

**Comment by dustyny** [comment]
Thank you @dianashk I was able to get a representative city (Miami) installed for my testing. Unfortunately it was to inaccurate for my use case. I was hoping to use Pelias as a secondary source of address correction and verification but there was to many interpolated addresses. Also the Geolocation that I get from Pelias was not close enough to what I get from my current provider so I could't match the records against that. 

That said the documentation is not clear on how to import data in. I spent a lot of time going through the code, documentation and trial and error. I'd recommend spending some time on that. I hit a brick wall pretty quickly and if I wasn't determined to test it, I would have stop much sooner. I think it's a real barrier for attracting contributors. 

I do have a side project that I'm going to try to use Pelias for. So I'll be back when I get some free time.
